prov/verbs/src/ep_rdm/verbs_ep_rdm.c

```diff
@@ -416,16 +419,19 @@ static int fi_ibv_rdm_ep_close(fid_t fid)
 				}
 			}
 		}
+		pthread_mutex_unlock(&av_entry->conn_lock);
 	}
 
         /* now destroy all connections that wasn't removed from HASH */
 	HASH_ITER(hh, ep->domain->rdm_cm->av_hash, av_entry, tmp) {
+		pthread_mutex_lock(&av_entry->conn_lock);
 		HASH_FIND(hh, av_entry->conn_hash, &ep,
 			  sizeof(struct fi_ibv_rdm_ep *), conn);
 		if (conn) {
 			HASH_DEL(av_entry->conn_hash, conn);
 			fi_ibv_rdm_conn_cleanup(conn);
```

Comment:
Could this connection cleanup be done outside the lock?

Reply:
Yes, that would be great.